src/transformState.js

```diff
@@ -5,7 +5,23 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (let i = 0; i < actions.length; i++) {
+    switch (actions[i].type) {
+      case 'addProperties':
+        Object.assign(state, actions[i].extraData);
+        break;
+      case 'removeProperties':
+        for (const action of actions[i].keysToRemove) {
+          delete state[action];
+        };
+        break;
+      case 'clear':
+        Object.keys(state).forEach(key => delete state[key]);
```

Comment:
It's one of the ways to delete all properties from an object. The other one is to assign an empty object to the `state` variable. In the second case, you'll create a new reference so they're not identical.

The first way is slower, creating a nested loop but keeping the same reference. The second way is faster, without a nested loop, but it creates a new reference.